src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -804,30 +823,57 @@ internal static string GetInputFile(PackArgs packArgs, IEnumerable<string> files
             string result;
 
-            candidates.RemoveAll(ext => ext.EndsWith(".json") && 
-                                    !ext.Equals(ProjectJsonPathUtilities.ProjectConfigFileName, StringComparison.OrdinalIgnoreCase) &&
-                                    !ext.EndsWith(ProjectJsonPathUtilities.ProjectConfigFileEnding, StringComparison.OrdinalIgnoreCase));
+            candidates.RemoveAll(ext => ext.EndsWith(".lock.json", StringComparison.OrdinalIgnoreCase) ||
+                                    (ext.EndsWith(".json", StringComparison.OrdinalIgnoreCase) && 
+                                    !Path.GetFileName(ext).Equals(ProjectJsonPathUtilities.ProjectConfigFileName, StringComparison.OrdinalIgnoreCase) &&
+                                    !ext.EndsWith(ProjectJsonPathUtilities.ProjectConfigFileEnding, StringComparison.OrdinalIgnoreCase)));
 
-            switch (candidates.Count)
+            if (!candidates.Any())
             {
-                case 1:
+                throw new ArgumentException(Strings.InputFileNotSpecified);
+            }
+            if (candidates.Count == 1)
+            {
+                result = candidates[0];
+            }
+            else
+            {
+                // Remove all nuspec files
+                candidates.RemoveAll(file => Path.GetExtension(file).Equals(NuGetConstants.ManifestExtension, StringComparison.OrdinalIgnoreCase));
+                if (candidates.Count == 1)
+                {
                     result = candidates[0];
-                    break;
-
-                case 2:
-                    // Remove all nuspec files
-                    candidates.RemoveAll(file => Path.GetExtension(file).Equals(NuGetConstants.ManifestExtension, StringComparison.OrdinalIgnoreCase));
+                }
+                else
+                {
+                    // Remove all json files
+                    candidates.RemoveAll(file => Path.GetExtension(file).Equals(".json", StringComparison.OrdinalIgnoreCase));
                     if (candidates.Count == 1)
                     {
                         result = candidates[0];
-                        break;
                     }
-                    goto default;
-                default:
-                    throw new ArgumentException(Strings.InputFileNotSpecified);
+                    else
+                    {
+                        throw new ArgumentException(Strings.InputFileNotSpecified);
+                    }
+                }
+            }
+
+            string fullPath = Path.GetFullPath(Path.Combine(packArgs.CurrentDirectory, result));
+            string directory = Path.GetDirectoryName(fullPath);
+
+            if (!directory.Equals(packArgs.CurrentDirectory, StringComparison.OrdinalIgnoreCase))
+            {
+                if (string.IsNullOrEmpty(packArgs.OutputDirectory))
+                {
+                    packArgs.OutputDirectory = packArgs.CurrentDirectory;
```

Comment:
This is the case where the user specified a filename but it's not in the current directory.  The other place is the case when the user specifies just a folder to pack.